dtm0/clk_src.c

```diff
@@ -0,0 +1,201 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_DTM0
+#include "lib/trace.h"
+
+#include "dtm0/clk_src.h"
+#include "lib/errno.h" /* EINVAL */
+#include "lib/assert.h" /* M0_ASSERT and so on */
+
+enum {
+	/* The hard limit in the now() busy loop.
+	 * The program should panic() if the physical
+	 * clock did not advance for too long (more than
+	 * MAX_RETRIES * ADV_NS nanoseconds).
+	 */
+	CS_PHYS_NOW_MAX_RETRIES = 16,
+
+	/* A soft limit in the now() busy loop.
+	 * This limit should be as small as possible but
+	 * big enough to make the clock advance.
+	 */
+	CS_PHYS_NOW_ADV_NS = 1,
+};
+
+struct m0_dtm0_clk_src_ops {
+	int (*cmp)(const struct m0_dtm0_ts *left,
```

Comment:
Prefixes for struct fields please, see coding-style.

Reply:
Done.